### PR TITLE
Adding the PyEvJosev module to replace the JsCarV2G module

### PR DESCRIPTION
**Pull request by SebaLukas**
Please test with this PR: [Integration of the EVCC from Josev into EVerest](https://github.com/EVerest/ext-switchev-iso15118/pull/13)

### CODE REVIEW

**Comment by SebaLukas** [comment]
Missing: The private contract cert key is not saved in the correct format (.pem) during the certificate installation process.

File: modules/PyEvJosev/module.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            m.implement_command(
+                'ev', cmd, getattr(self, f'_handler_{cmd}'))
+            
+        # init ready event
+        self._ready_event = threading.Event()
+
+        self._mod = m
+        self._mod.init_done(self._ready)
+
+    def start_evcc_hander(self):
```
Comment (sdrabb) [inline modules/PyEvJosev/module.py:80]: typo : )

**Comment by sdrabb** [comment]
@SebaLukas what about EVSlac side? Are you working on it?

**Comment by SebaLukas** [comment]
@sdrabb we are working on it :)

**Comment by SebaLukas** [comment]
For PnC testing I will provide a extra pnc config

**Review by sdrabb — APPROVED** [review]

File: modules/PyEvJosev/module.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+# fmt: off
+JOSEV_WORK_DIR = Path(__file__).parent / '../../3rd_party/josev'
+sys.path.append(JOSEV_WORK_DIR.as_posix())
+
+from iso15118.evcc import EVCCHandler
+from iso15118.evcc.controller.simulator import SimEVController
+from iso15118.evcc.evcc_config import EVCCConfig
+from iso15118.evcc.everest import context as JOSEV_CONTEXT
+from iso15118.shared.exificient_exi_codec import ExificientEXICodec
+from iso15118.shared.settings import set_PKI_PATH
```
Comment (a-w50) [inline modules/PyEvJosev/module.py:19]: I would prefer having this solved differently / we need to come up with a best practice how external python modules are distributed to the system
- either by copying them directly
- or by pip
- wheels?

File: modules/PyEvJosev/utilities.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    def emit(self, record):
+        msg = self.format(record)
+
+        log_level: int = record.levelno
+        if log_level == logging.CRITICAL:
+            log.critical(msg)
+        elif log_level == logging.ERROR:
+            log.error(msg)
+        elif log_level == logging.WARNING:
```
Comment (a-w50) [inline modules/PyEvJosev/utilities.py:24]: This is, I guess, exactly the same as in the EVSE part.  I'd prefer removing this duplication by creating a shared library for this inside 'everest-core/lib' - again we would need to think on how to package and distribute/install that later on

**Review by a-w50 — COMMENTED** [review]
See my comments, depending on if we want to postpone these issues or not, this PR can be approved or not

File: modules/PyEvJosev/utilities.py (hunk 3 above)
Comment (SebaLukas) [reply modules/PyEvJosev/utilities.py:24]: It's the same code

**Comment by SebaLukas** [comment]
Those are both good points. However, I would not solve them in this PR, but first discuss them and then possibly solve them in a separate PR.

**Review by a-w50 — APPROVED** [review]
Fine, please rebase and squash

**Comment by corneliusclaussen** [comment]
Converted to draft, SIL is not working at the moment for ISO. This branch should fix SIL before merging

**Pull request merged**